src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -0,0 +1,380 @@
+        // software keyboard when the control gains focus and dismiss it when the control loses focus.
+        // TODO GitHub #3639: Should Input Pane display policy be Automatic
+        _editContext.InputPaneDisplayPolicy(Core::CoreTextInputPaneDisplayPolicy::Manual);
+
+        // set the input scope to Text because this control is for any text.
+        _editContext.InputScope(Core::CoreTextInputScope::Text);
+
+        _editContext.TextRequested({ this, &TSFInputControl::_textRequestedHandler });
+
+        _editContext.SelectionRequested({ this, &TSFInputControl::_selectionRequestedHandler });
+
+        _editContext.FocusRemoved({ this, &TSFInputControl::_focusRemovedHandler });
+
+        _editContext.TextUpdating({ this, &TSFInputControl::_textUpdatingHandler });
+
+        _editContext.SelectionUpdating({ this, &TSFInputControl::_selectionUpdatingHandler });
+
+        _editContext.FormatUpdating({ this, &TSFInputControl::_formatUpdatingHandler });
+
+        _editContext.LayoutRequested({ this, &TSFInputControl::_layoutRequestedHandler });
+
+        _editContext.CompositionStarted({ this, &TSFInputControl::_compositionStartedHandler });
+
+        _editContext.CompositionCompleted({ this, &TSFInputControl::_compositionCompletedHandler });
+    }
+
+    // Method Description:
+    // - NotifyFocusEnter handler for notifying CoreEditTextContext of focus enter
+    //   when TerminalControl receives focus.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::NotifyFocusEnter()
+    {
+        if (_editContext != nullptr)
+        {
+            _editContext.NotifyFocusEnter();
+        }
+    }
+
+    // Method Description:
+    // - NotifyFocusEnter handler for notifying CoreEditTextContext of focus leaving.
+    //   when TerminalControl no longer has focus.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TSFInputControl::NotifyFocusLeave()
+    {
+        if (_editContext != nullptr)
+        {
+            // _editContext.NotifyFocusLeave(); TODO GitHub #3645: Enabling causes IME to no longer show up, need to determine if required
+        }
+    }
+
+    // Method Description:
+    // - Scales a Rect based on a scale factor
+    // Arguments:
+    // - rect: Rect to scale by scale
+    // - scale: amount to scale rect by
+    // Return Value:
+    // - Rect scaled by scale
+    inline Rect ScaleRect(Rect rect, double scale)
+    {
+        const float scaleLocal = gsl::narrow<float>(scale);
+        rect.X *= scaleLocal;
+        rect.Y *= scaleLocal;
+        rect.Width *= scaleLocal;
+        rect.Height *= scaleLocal;
+        return rect;
+    }
+
+    // Method Description:
+    // - Converts a COLORREF to Color
+    // Arguments:
+    // - colorref: COLORREF to convert to Color
+    // Return Value:
+    // - Color containing the RGB values from colorref
+    inline winrt::Windows::UI::Color ColorRefToColor(const COLORREF& colorref)

```

Comment:
Yea let's definitely move this to `WinRTUtils` - I'd probably just make a `Utils.h/.cpp` in that project for this.

Reply:
I've created a Utils.h and moved ColorRefToColor and ScaleRect there.  Will be in next code push.